.github/workflows/docs.yml

```diff
@@ -0,0 +1,37 @@
+name: Generate docs
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'src/**'
+      - 'README.md'
+    branches: [main]
+
+jobs:
+  generate-docs:
+    if: github.repository_owner == 'viamrobotics'
+    runs-on: [self-hosted, x64]
+    container:
+      image: ghcr.io/viamrobotics/canon:amd64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Generate docs
+        run: doxygen Doxyfile
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: html-docs
+          path: etc/docs/api/current
```

Comment:
discussed IRL - I'm not 100% sure if we need this - @njooma do you know?

Reply:
I'm not sure if it's needed either. Did you test without this? Does it work?

Reply:
I'm not sure! I don't really know how to test this without it, since it's not yet merged so the branch that the job actually runs from doesn't know about this job.

I think the stakes are pretty low here though, so if y'all are comfortable I might just merge and then we can iterate from there if anything isn't working right.

Reply:
yeah I'm okay merging as is - I kept it for TS doc generation as well - was more curious than anything else.